cypress/e2e/environment.cy.js

```diff
@@ -0,0 +1,389 @@
+  cy.url().should("eq", Cypress.config().baseUrl + "/console/");
+  cy.get('[aria-label="Environment card"]')
+    .contains(testProjectName(number))
+    .should("not.exist");
+};
+
+describe("Environment", function () {
+  it("1.1 cancel creation of an environment", function () {
+    cy.visit("/console/");
+    cy.get(newEnvCard).click();
+    cy.url().should(
+      "eq",
+      Cypress.config().baseUrl + "/console/environment/create"
+    );
+    createEnv(1);
+    cy.get("button").contains("Cancel").click();
+    cy.url().should("eq", Cypress.config().baseUrl + "/console/");
+  });
+
+  it("1.2 Create ne environment", function () {
+    //fill the form and submit
+    cy.visit("/console/environment/create");
+    createEnv(2, false);
+    cy.get("button").contains("Submit").should("be.disabled");
+    cy.get(nameInput).type(testName(2));
+    cy.get("button").contains("Submit").should("not.be.disabled");
+    cy.get("button").contains("Submit").click();
+    cy.url().should("contain", "/console/lsm/catalog?env=");
+    //go back to gome and check if env is visible
+    cy.get(".pf-c-breadcrumb__item").first().click();
+    cy.url().should("eq", Cypress.config().baseUrl + "/console/");
+    cy.get('[aria-label="Environment card"]').should(
+      "any.contain",
+      testName(2)
+    );
+  });
+
+  it("1.3 delete an environment", function () {
+    //Fill The form and submit
+    cy.visit("/console/environment/create");
+    createEnv(3, true, true);
+    cy.get("button").contains("Submit").click();
+    cy.url().should("contain", "/console/lsm/catalog?env=");
+    //Go to settings
+    cy.get('[aria-label="Settings actions"]').click();
+    cy.url().should("contain", "/console/settings?env=");
+    cy.get('[aria-label="Name-value"]').should("contain", testName(3));
+    deleteEnv(3);
+  });
+
+  it("1.4 Edit created environment", function () {
+    //Fill The form and submit
+    cy.visit("/console/environment/create");
+    createEnv(4, true, true);
+    cy.get("button").contains("Submit").click();
+    //Go to settings
+    cy.get('[aria-label="Settings actions"]').click();
+    cy.get('[aria-label="Name-value"]').should("contain", testName(4));
+    cy.wrap(parameterTypes).each((parameter) => {
+      if (parameter.name !== "Repository" && parameter.name !== "Branch") {
+        cy.get('[aria-label="' + parameter.name + '-toggle-edit"]').click({
+          force: true,
+        });
+      }
+
+      switch (parameter.type) {
+        case "text":
+          cy.get('[aria-label="' + parameter.name + '-input"]')
+            .clear()
+            .type("New Value " + parameter.name);
+          break;
+        case "icon":
+          cy.get(iconInput).attachFile(icon);
+          break;
+        case "typeahead":
+          cy.get('[aria-label="Project Name-typeahead"]')
+            .clear()
+            .type("New Value " + parameter.name);
+
+          cy.get('[aria-label="Project Name-select-input"]').then(($list) => {
```

Comment:
why the dollar sign in your variable?

Reply:
That's the convention cypress used in the docs to mark html components, at least in snippets they have, I followed, but I can change it, as I don't mind :)

Reply:
I don't mind it, I just never saw that. That's why I was wondering what it meant

Reply:
Agreed to follow the code guidelines from cypress and use the dollar sign to specify HTML elements and components.